### ISSUE TITLE
How to do with variable length sequence in tensorflow 2.0 docs example "Neural Machine Translation with Attention"

### ISSUE THREAD

**Issue by hamelsmu** [hamelsmu/Issue-Label-Bot-Examples#108]
The following code in the example is just for the fix length, how to deal with the variable length(such as the valid length is 7, 6, 8), Maybe it can deal with tf.keras.layers.Masking, is it any other methods?
```
class Encoder(tf.keras.Model):
  def __init__(self, vocab_size, embedding_dim, enc_units, batch_sz):
    super(Encoder, self).__init__()
    self.batch_sz = batch_sz
    self.enc_units = enc_units
    self.embedding = tf.keras.layers.Embedding(vocab_size, embedding_dim)
    self.gru = tf.keras.layers.GRU(self.enc_units, 
                                   return_sequences=True, 
                                   return_state=True, 
                                   recurrent_initializer='glorot_uniform')

  def call(self, x, hidden):
    x = self.embedding(x)
    output, state = self.gru(x, initial_state = hidden)        
    return output, state

  def initialize_hidden_state(self):
    return tf.zeros((self.batch_sz, self.enc_units))
```

Note testing `https://github.com/tensorflow/tensorflow/issues/27642`
